marketplace-builder/src/hooks.rs

```diff
@@ -0,0 +1,104 @@
+use async_trait::async_trait;
+use espresso_types::v0_3::BidTxBody;
+
+use espresso_types::SeqTypes;
+use hotshot::types::EventType;
+
+use hotshot::types::Event;
+
+use marketplace_builder_core::service::BuilderHooks;
+
+use espresso_types::FeeAmount;
+
+use espresso_types::eth_signature_key::EthKeyPair;
+
+use espresso_types::NamespaceId;
+
+use hotshot_types::traits::node_implementation::NodeType;
+
+use surf_disco::Client;
+
+use tide_disco::Url;
+use tracing::error;
+use tracing::info;
+
+pub async fn connect_to_solver(
+    solver_api_url: Url,
+) -> Option<Client<hotshot_events_service::events::Error, <SeqTypes as NodeType>::Base>> {
+    let client = Client::<hotshot_events_service::events::Error, <SeqTypes as NodeType>::Base>::new(
+        solver_api_url.join("v0/marketplace-solver/").unwrap(),
+    );
+
+    if !(client.connect(None).await) {
+        return None;
+    }
+
+    tracing::info!("Builder client connected to the solver api");
+
+    Some(client)
+}
+
+/// Builder hooks for espresso sequencer
+/// Provides bidding and transaction filtering on top of base builder functionality
+pub(crate) struct EspressoHooks {
+    /// ID of namespace to filter and bid for
+    pub(crate) namespace_id: NamespaceId,
```

Comment:
In the previous implementation, this is an `Option<Vec<...>>`. Did we decide to always filter one namespace for this version?

Reply:
I don't think it was a `Vec` before, but I do plan to make it a `Vec` as it's quite an easy change and gives us more flexibility
For `Option` part, it was first there because originally it was supposed to be merged into hotshot-builder-core and the idea was that when running legacy builder we'd want to set it to None and filter nothing. I imagine for generic builder we'd want to make it an Option again